### PR TITLE
loader, cfg: choose region based on language if multiple regions available

### PR DESCRIPTION
**Pull request by wwylele**
Mainly to deal with games like pokemon which is marked "region free"  in the header.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4232)

### CODE REVIEW

File: src/core/hle/service/cfg/cfg.cpp [hunk 1]
```diff
@@ -571,7 +572,8 @@ Module::Module() {
 Module::~Module() = default;
 
 /// Checks if the language is available in the chosen region, and returns a proper one
-static SystemLanguage AdjustLanguageInfoBlock(u32 region, SystemLanguage language) {
+static std::tuple<u32 /*region*/, SystemLanguage> AdjustLanguageInfoBlock(
```
Comment (zhaowenlan1779) [inline src/core/hle/service/cfg/cfg.cpp]: Any particular reason this is not a `std::pair`?

File: src/core/hle/service/cfg/cfg.cpp [hunk 2]
```diff
@@ -571,7 +572,8 @@ Module::Module() {
...
+    const std::vector<u32>& region_code, SystemLanguage language) {
```
Comment (zhaowenlan1779) [inline src/core/hle/service/cfg/cfg.cpp]: ultra minor nit, but probably name it `region_codes`...

File: src/core/hle/service/cfg/cfg.cpp [hunk 3]
```diff
@@ -590,21 +592,28 @@ static SystemLanguage AdjustLanguageInfoBlock(u32 region, SystemLanguage languag
...
-    return language;
+    // The language is not available in any available region, so default to the first region and
+    // language
+    u32 default_region = region_code[0];
+    return {default_region, region_languages[default_region][0]};
 }
 
-void Module::SetPreferredRegionCode(u32 region_code) {
-    preferred_region_code = region_code;
+void Module::SetPreferredRegionCodes(const std::vector<u32>& region_code) {
```
Comment (zhaowenlan1779) [inline src/core/hle/service/cfg/cfg.cpp]: ditto

File: src/core/hle/service/cfg/cfg.cpp (hunk 1 above)
Comment (wwylele) [reply src/core/hle/service/cfg/cfg.cpp]: no particular reason... I just like to use tuple more for unified code style

**Pull request merged**